statement/parser_test.go

```diff
@@ -147,11 +154,90 @@ func TestSplitStatements(t *testing.T) {
 	runSplitStatement(t, "SELECT 1; SELECT 2; SELECT 3; SELECT 4; SELECT 5; SELECT 6", []string{"SELECT 1", " SELECT 2", " SELECT 3", " SELECT 4", " SELECT 5", " SELECT 6"})
 }
 
-func TestValueToNamedValue(t *testing.T) {
-	utils.AssertEqual(len(valueToNamedValue([]driver.Value{})), 0, t, "valueToNamedValue of empty array is wrong")
+func runValidateSetStatement(t *testing.T, value string, shouldError bool, messagePart string) {
+	res := validateSetStatement(value)
+	if !shouldError && res != nil {
+		t.Errorf("validateSetStatement returned an error, but it shouldn't for: %s", value)
+		return
+	}
+	if shouldError && !strings.Contains(res.Error(), messagePart) {
+		t.Errorf("validateSetStatement returned an error '%s', but it should contain '%s'", res.Error(), messagePart)
+	}
+}
```

Comment:
I get what you're trying to do, go around go's verbosity and avoid repeating if err... , but this makes it hard to read tests.
I'd personally split this into two helper methods - `validateSetStatementSuccess` and `validateSetStatementFailure`
Same `runPrepareQuery`, it would improve the tests considerably.